app/src/main/java/com/plenart/organizeme/activities/MainActivity.kt

```diff
@@ -3,188 +3,95 @@ package com.plenart.organizeme.activities
 import android.app.Activity
 import android.content.Intent
 import android.os.Bundle
+import android.os.Handler
+import android.os.Looper
 import android.util.Log
-import android.view.MenuItem
 import android.view.View
-import android.widget.TextView
-import androidx.core.view.GravityCompat
-import androidx.recyclerview.widget.LinearLayoutManager
-import com.bumptech.glide.Glide
-import com.google.android.material.navigation.NavigationView
+import androidx.activity.viewModels
+import androidx.appcompat.app.AppCompatActivity
+import androidx.appcompat.widget.Toolbar
+import androidx.navigation.NavController
+import androidx.navigation.fragment.NavHostFragment
+import androidx.navigation.ui.setupActionBarWithNavController
 import com.google.firebase.auth.FirebaseAuth
 import com.plenart.organizeme.R
-import com.plenart.organizeme.adapters.BoardItemsAdapter
 import com.plenart.organizeme.databinding.ActivityMainBinding
-import com.plenart.organizeme.databinding.AppBarMainBinding
-import com.plenart.organizeme.databinding.MainContentBinding
 import com.plenart.organizeme.firebase.Firestore
-import com.plenart.organizeme.interfaces.BoardItemClickInterface
-import com.plenart.organizeme.models.Board
-import com.plenart.organizeme.utils.Constants
-import de.hdodenhof.circleimageview.CircleImageView
+import com.plenart.organizeme.viewModels.MainActivityViewModel
 
-class MainActivity : BaseActivity(), NavigationView.OnNavigationItemSelectedListener {
+class MainActivity : AppCompatActivity() {
 
-    private lateinit var mainActivityBinding: ActivityMainBinding;
-    private lateinit var appBarMainBinding: AppBarMainBinding;
-    private lateinit var mainContentBinding: MainContentBinding;
-
-    private lateinit var mUserName: String;
+    private lateinit var binding: ActivityMainBinding
+    private val viewModel: MainActivityViewModel by viewModels()
 
+    private lateinit var toolbar: Toolbar
+    private lateinit var navController: NavController
 
     override fun onCreate(savedInstanceState: Bundle?) {
+        setTheme(R.style.AppTheme)
         super.onCreate(savedInstanceState)
-        mainActivityBinding = ActivityMainBinding.inflate(layoutInflater)
-        setContentView(mainActivityBinding.root)
-
-        setUpActionBar();
-        mainActivityBinding.navView.setNavigationItemSelectedListener(this);
-
-        Firestore().loadUserData(this,true);
 
-        appBarMainBinding.fabCreateBoard.setOnClickListener{
-            val intent = Intent(this,CreateBoardActivity::class.java);
-            intent.putExtra(Constants.NAME, mUserName);
+        binding = ActivityMainBinding.inflate(layoutInflater)
+        setContentView(binding.root)
```

Comment:
I think this can be replaced with `binding = DataBindingUtils.setContentView()`

Reply:
Yes, if I were to use dataBinding. Since I'm using viewBinding, as far as I know, this is the correct way.